### ISSUE TITLE
[Bug]: ControlNet via API doesn't work for model based on 2.1, but does work via web-ui

### ISSUE THREAD

**Issue by Yelorix** [Mikubill/sd-webui-controlnet#696]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### What happened?

I am able to use ControlNet openpose models for models based on Stable Diffusion 1.5 and also 2.1 (using the model trained by Thibaud). This works as expected, I load the model and change the config file setting and it just works.

But if I use the txt2img API endpoint (sdapi/v1/txt2img), the output image is only influenced by ControlNet if I use the model that's based on Stable Diffusion 1.5

If I use the exact same API call (the only difference being the name of the controlnet model), while I have a model based on Stable Diffusion 2.1 loaded, I just get a generic result that represents the prompt but not the control input image.

I get that this is probably something that I'm doing wrong, because I probably wouldn't be the first one to mention this bug if I wasn't.

There are no errors, and the command prompt shows that the model is correctly being loaded. I also notice that the generation time is longer than when I'm not using ControlNet. So I'm assuming that something's happening still.

![image](https://user-images.githubusercontent.com/77750629/229309681-36a13f45-04cc-49bc-9e5d-1fe719d822f5.png)

### Steps to reproduce the problem

I'm not able to test this on multiple devices, so I don't know if it's reproducable. But It's consistently not working for me.

1. Make an API call to Automatic1111's txt2img endpoint using a 1.5 model (stable diffusion model AND controlnet model)
2. Make the same API call but now use a 2.1 model (stable diffusion AND controlnet model)

I used these parameters:

{"prompt":"Man in red overalls, white background","styles":[],"seed":-1,"subseed":-1,"seed_resize_from_h":-1,"seed_resize_from_w":-1,"sampler_name":"Euler a","batch_size":1,"n_iter":1,"steps":10,"cfg_scale":4.0,"width":768,"height":768,"restore_faces":true,"s_noise":1.0,"override_settings":{},"alwayson_scripts":{"controlnet":{"args":[{"input_image":"_the image as base64 string_","module":"none","model":"openpose-sd21-safe","weight":1,"guidance_end":1.0}]}}}

### What should have happened?

I expect the same behaviour in the web-ui as via the API, the output being influenced by the control image.

### Commit where the problem happens

webui: 22bcc7be428c94e9408f589966c2040187245d81
controlnet: 241c05f8 (Thu Mar 23 15:18:35 2023)

### What browsers do you use to access the UI ?

Google Chrome

### Command Line Arguments

```Shell
--api
```

### Console logs

```Shell
venv "C:\Users\MyName\stable-diffusion-webui\venv\Scripts\Python.exe"
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Commit hash: 22bcc7be428c94e9408f589966c2040187245d81
Installing requirements for Web UI

Launching Web UI with arguments: --api
No module 'xformers'. Proceeding without it.
==============================================================================
You are running torch 1.12.1+cu113.
The program is tested to work with torch 1.13.1.
To reinstall the desired version, run with commandline flag --reinstall-torch.
Beware that this will cause a lot of large files to be downloaded, as well as
there are reports of issues with training tab on the latest version.

Use --skip-version-check commandline argument to disable this check.
==============================================================================
Loading weights [cae1bee30e] from C:\Users\MyName\stable-diffusion-webui\models\Stable-diffusion\illuminatiDiffusionV1_v11.safetensors
Creating model from config: C:\Users\MyName\stable-diffusion-webui\models\Stable-diffusion\illuminatiDiffusionV1_v11.yamlLatentDiffusion: Running in v-prediction mode
DiffusionWrapper has 865.91 M params.
Applying cross attention optimization (Doggettx).
Textual inversion embeddings loaded(3): embellish1, embellish2, embellish3
Textual inversion embeddings skipped(1): HyperStylizeVA
Model loaded in 15.9s (load weights from disk: 0.5s, create model: 0.3s, apply weights to model: 11.0s, apply half(): 1.3s, move model to device: 1.2s, load textual inversion embeddings: 1.5s).
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 30.4s (import torch: 4.0s, import gradio: 2.4s, import ldm: 1.4s, other imports: 1.6s, setup codeformer: 0.3s, load scripts: 3.1s, load SD checkpoint: 16.1s, create ui: 1.2s, gradio launch: 0.2s, scripts app_started_callback: 0.1s).
Loading model: openpose-sd21 [64de50ad]
Loaded state_dict from [C:\Users\MyName\stable-diffusion-webui\extensions\sd-webui-controlnet\models\openpose-sd21.ckpt]
ControlNet model openpose-sd21 [64de50ad] loaded.
Loading preprocessor: none
100%|██████████████████████████████████████████████████| 10/10 [00:28<00:00,  2.90s/it]
Total progress: 100%|██████████████████████████████████| 10/10 [00:23<00:00,  2.63s/it]
```

### Additional information

I have used a different model for 2.1 to try that out as well, and that also didn't work.

The thing about this issue is, is that it only occurs via the API call, and only for the 2.1 model. The same API call on 1.5 works, and the same prompt/settings in the webui on 2.1 also work. So my ControlNet extension works, my Automatic1111 web-ui works, the API call I'm making is formatted correctly (or at least correctly enough so that it works for the 1.5 models!).

I can only deduce that there's something missing in the code triggered by the API call that isn't missing when you start the generation from the web-ui. The only thing I could think off was the config file, so I've already tried renaming the config file to the model, or deleting all config files except the one that I needed to be sure that it's that one getting used. And it made no difference. I've also tried to delete the config file to see if I'd get an error and as expected: I got an error. So that's where my debugging stopped.

**Comment by Yelorix** [comment]
![image](https://user-images.githubusercontent.com/77750629/229314447-7fe4686f-2c61-4622-bd62-2070e03f21b1.png)

I have found that there is no bug. The issue I had was that I thought the output was not influenced by ControlNet, but it was just that the weight was too low. I was not thinking about that because the API default is 0.5 and the web-ui default is 1.

The reason this stood out to me as a bug, was because the openpose model for 1.5 is a lot stronger than the one I have for 2.1, so at the default 0.5 value, these outputs clearly showed the pose. And the 2.1 performed just as well at the default 1 weight in the web-ui, but failed to show the pose at the API default weight of 0.5

Shame on me for doubting this great extension!
Thanks for your work.
